datalad/tests/test_cmd.py

```diff
@@ -0,0 +1,43 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Test command call wrapper
+"""
+
+import os
+
+import nose.tools
+
+from datalad.cmd2 import Runner
+from datalad.tests.utils import with_tempfile, assert_cwd_unchanged
+
+
+@assert_cwd_unchanged
+@with_tempfile
+def test_runner_dry(tempfile):
+
+    runner = Runner(dry=True)
+    cmd = 'echo Testing dry run > %s' % tempfile
+    ret = runner.run(cmd)
+    nose.tools.assert_is(None, ret, "Dry run of: %s resulted in exitcode %s" % (cmd, ret))
```

Comment:
Of course. I'm just making use of completion feature to know what's
available and at the end importing the ones I actually used ;-)

2015-03-08 4:41 GMT+01:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/tests/test_cmd.py
> <https://github.com/datalad/datalad/pull/53#discussion_r26002001>:
>
> > +import os
> > +
> > +import nose.tools
> > +
> > +from datalad.cmd2 import Runner
> > +from datalad.tests.utils import with_tempfile, assert_cwd_unchanged
> > +
> > +
> > +@assert_cwd_unchanged
> > +@with_tempfile
> > +def test_runner_dry(tempfile):
> > +
> > +    runner = Runner(dry=True)
> > +    cmd = 'echo Testing dry run > %s' % tempfile
> > +    ret = runner.run(cmd)
> > +    nose.tools.assert_is(None, ret, "Dry run of: %s resulted in exitcode %s" % (cmd, ret))
>
> better (would be mroe consistent with how we do elsewhere) do
>
> from nose.tools import assert_is, assert_equal
>
> and use them in the code without nose.tools prefix
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/53/files#r26002001>.
>